pandas/core/arrays/datetimes.py

```diff
@@ -1235,11 +1235,9 @@ def _generate_regular_range(cls, start, end, periods, freq):
         tz = None
         if isinstance(start, Timestamp):
             tz = start.tz
-            start = start.to_pydatetime()
```

Comment:
This conversion ``to_pydatetime`` seemed to be causing several tests in ``tests/test_resample.py`` to fail because the nanosecond precision was lost. When I removed this it fixed the problem and didn't seem to break any other tests locally.